snips-nlu-ontology-parsers/src/builtin_entity_parser.rs

```diff
@@ -37,11 +49,82 @@ impl BuiltinEntityParser {
         &self,
         sentence: &str,
         filter_entity_kinds: Option<&[BuiltinEntityKind]>,
+    ) -> Vec<BuiltinEntity> {
+        if NON_SPACE_SEPARATED_LANGUAGES.contains(&self.lang) {
+            let original_tokens_bytes_ranges: Vec<Range<usize>> = NON_SPACE_REGEX
+                .find_iter(sentence)
+                .map(|m| Range { start: m.start(), end: m.end() })
+                .collect();
+
+            let joined_sentence = original_tokens_bytes_ranges
+                .iter()
+                .map(|r| &sentence[r.start..r.end])
+                .join("");
+
+            if original_tokens_bytes_ranges.len() == 0 {
+                return vec![];
+            }
+
+            let joined_sentence_match_end_byte_index_to_token_index = HashMap::<usize, usize>::from_iter(
+                original_tokens_bytes_ranges
+                    .iter()
+                    .enumerate()
+                    .fold(vec![], |mut acc: Vec<(usize, usize)>, (i, ref r)| {
```

Comment:
Perhaps replace `i` with `token_index` and `r` with something like `original_range` to  clarify a bit.